src/App.tsx

```diff
@@ -23,7 +32,22 @@ export function App() {
     darkMode
       ? document.querySelector('html')?.classList.add('dark')
       : document.querySelector('html')?.classList.remove('dark');
-  }, [darkMode]);
+
+    const resolvedLanguage = resolveLanguage(company?.settings.language_id);
+
+    if (resolvedLanguage?.locale) {
+      if (!i18n.hasResourceBundle(resolvedLanguage.locale, 'translation')) {
+        import(
+          `./resources/lang/${resolvedLanguage.locale}/${resolvedLanguage.locale}.json`
+        ).then((response) => {
```

Comment:
The reason for dynamically importing json files as translation resources is actually memory. If we create some index file and include them all at once, the index.tsx file will be as heavy as `10MB` and will fail in `npm run build` command because it is over `1500KB`. I did some research on Google and almost all people do it this way. You can check out the first commit on this branch to see how I implemented this using all the resource files immidiatelly.